### PR TITLE
feat(accel_brake_map_calibrator): replace polling takeData function with the callback function

### PR DESCRIPTION
**Pull request by N-Eiki**
## Description

In this PR, instead of getting msg in callback, it is changed to get msg using takeData function of polling_subscriber in utils.

## Tests performed
```
// terminal1
ros2 launch accel_brake_map_calibrator accel_brake_map_calibrator.launch.xml rviz:=true use_sim_time:=true

// terminal2
ros2 bag play <rosbag file> --clock
```
The module works as before. By the way, I only have a rosbag file that was collected with the old messages (autoware_auto_* msgs). I don't have any rosbag files with the new messages (autoware_* msgs). So, I confirmed with the old rosbag file and replaced the messages later.

### before with old msg

https://github.com/autowarefoundation/autoware.universe/assets/53928021/6f2251a7-ff99-4843-b2a9-4b2ec788da77

### after with old msg

https://github.com/autowarefoundation/autoware.universe/assets/53928021/0af77938-8e45-491c-84c3-5d87272128a4

## Interface changes
none

## Effects on system behavior
none

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].
- [x] The PR has been properly tested.
- [x] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: vehicle/accel_brake_map_calibrator/include/accel_brake_map_calibrator/accel_brake_map_calibrator_node.hpp [hunk 1]
```diff
@@ -1,5 +1,5 @@
 //
-// Copyright 2020 Tier IV, Inc. All rights reserved.
+// Copyright 2024 Tier IV, Inc. All rights reserved.
```
Comment (tkimura4) [inline vehicle/accel_brake_map_calibrator/include/accel_brake_map_calibrator/accel_brake_map_calibrator_node.hpp:2]: There is no need to change the copyright notice.

File: vehicle/accel_brake_map_calibrator/include/accel_brake_map_calibrator/accel_brake_map_calibrator_node.hpp [hunk 2]
```diff
@@ -56,8 +57,8 @@
 namespace accel_brake_map_calibrator
 {
 
-using autoware_vehicle_msgs::msg::SteeringReport;
-using autoware_vehicle_msgs::msg::VelocityReport;
+using autoware_auto_vehicle_msgs::msg::SteeringReport;
+using autoware_auto_vehicle_msgs::msg::VelocityReport;
```
Comment (tkimura4) [inline vehicle/accel_brake_map_calibrator/include/accel_brake_map_calibrator/accel_brake_map_calibrator_node.hpp:61]: We will not use autoware_auto_msgs in the future, so these changes are unnecessary.

File: vehicle/accel_brake_map_calibrator/include/accel_brake_map_calibrator/accel_brake_map_calibrator_node.hpp [hunk 3]
```diff
@@ -23,12 +23,13 @@
...
 #include "tier4_autoware_utils/ros/logger_level_configure.hpp"
+#include "tier4_autoware_utils/ros/polling_subscriber.hpp"
 #include "tier4_autoware_utils/ros/transform_listener.hpp"
 
 #include <Eigen/Dense>
 
-#include "autoware_vehicle_msgs/msg/steering_report.hpp"
-#include "autoware_vehicle_msgs/msg/velocity_report.hpp"
+#include "autoware_auto_vehicle_msgs/msg/steering_report.hpp"
+#include "autoware_auto_vehicle_msgs/msg/velocity_report.hpp"
```
Comment (tkimura4) [inline vehicle/accel_brake_map_calibrator/include/accel_brake_map_calibrator/accel_brake_map_calibrator_node.hpp:32]: We will not use autoware_auto_msgs in the future, so these changes are unnecessary.

File: vehicle/accel_brake_map_calibrator/include/accel_brake_map_calibrator/accel_brake_map_calibrator_node.hpp (hunk 1 above)
Comment (N-Eiki) [reply vehicle/accel_brake_map_calibrator/include/accel_brake_map_calibrator/accel_brake_map_calibrator_node.hpp:2]: Ok. I undo this change!

File: vehicle/accel_brake_map_calibrator/package.xml [hunk 4]
```diff
@@ -7,6 +7,7 @@
   <maintainer email="tomoya.kimura@tier4.jp">Tomoya Kimura</maintainer>
   <maintainer email="taiki.tanaka@tier4.jp">Taiki Tanaka</maintainer>
   <maintainer email="takeshi.miura@tier4.jp">Takeshi Miura</maintainer>
+  <maintainer email="eiki.nagata@tier4.jp">Eiki Nagata</maintainer>
```
Comment (shmpwk) [inline vehicle/accel_brake_map_calibrator/package.xml:10]: :100:

File: vehicle/accel_brake_map_calibrator/src/accel_brake_map_calibrator_node.cpp [hunk 5]
```diff
@@ -46,10 +46,10 @@ AccelBrakeMapCalibrator::AccelBrakeMapCalibrator(const rclcpp::NodeOptions & nod
   max_accel_ = declare_parameter<double>("max_accel", 5.0);
   min_accel_ = declare_parameter<double>("min_accel", -5.0);
   pedal_to_accel_delay_ = declare_parameter<double>("pedal_to_accel_delay", 0.3);
-  max_data_count_ = declare_parameter<int>("max_data_count", 200);
+  max_data_count_ = static_cast<int>(declare_parameter<int>("max_data_count", 200));
```
Comment (shmpwk) [inline vehicle/accel_brake_map_calibrator/src/accel_brake_map_calibrator_node.cpp:49]: What do you intend to use static_cast?

File: vehicle/accel_brake_map_calibrator/src/accel_brake_map_calibrator_node.cpp [hunk 6]
```diff
@@ -46,10 +46,10 @@ AccelBrakeMapCalibrator::AccelBrakeMapCalibrator(const rclcpp::NodeOptions & nod
...
   pedal_accel_graph_output_ = declare_parameter<bool>("pedal_accel_graph_output", false);
   progress_file_output_ = declare_parameter<bool>("progress_file_output", false);
-  precision_ = declare_parameter<int>("precision", 3);
+  precision_ = static_cast<int>(declare_parameter<int>("precision", 3));
```
Comment (shmpwk) [inline vehicle/accel_brake_map_calibrator/src/accel_brake_map_calibrator_node.cpp:52]: What do you intend to use static_cast?

File: vehicle/accel_brake_map_calibrator/src/accel_brake_map_calibrator_node.cpp [hunk 7]
```diff
@@ -269,15 +253,55 @@ bool AccelBrakeMapCalibrator::getCurrentPitchFromTF(double * pitch)
...
+  debug_values_.data.at(CURRENT_RAW_PITCH) = static_cast<float>(raw_pitch);
   *pitch = lowpass(*pitch, raw_pitch, 0.2);
-  debug_values_.data.at(CURRENT_PITCH) = *pitch;
+  debug_values_.data.at(CURRENT_PITCH) = static_cast<float>(*pitch);
   return true;
 }
 
-void AccelBrakeMapCalibrator::timerCallback()
+bool AccelBrakeMapCalibrator::fetch_data()
+
```
Comment (shmpwk) [inline vehicle/accel_brake_map_calibrator/src/accel_brake_map_calibrator_node.cpp:267]:
```suggestion
```

File: vehicle/accel_brake_map_calibrator/src/accel_brake_map_calibrator_node.cpp [hunk 8]
```diff
@@ -269,15 +253,55 @@ bool AccelBrakeMapCalibrator::getCurrentPitchFromTF(double * pitch)
...
+{
+  // take data from subscribers
+  // take actuation data
+  ActuationStatusStamped::ConstSharedPtr actuation_status_ptr = actuation_status_sub_.takeData();
+  ActuationCommandStamped::ConstSharedPtr actuation_cmd_ptr = actuation_cmd_sub_.takeData();
+  if (actuation_status_ptr) {
+    take_actuation_status(actuation_status_ptr);
+  } else if (actuation_cmd_ptr) {
+    take_actuation_command(actuation_cmd_ptr);
+
```
Comment (shmpwk) [inline vehicle/accel_brake_map_calibrator/src/accel_brake_map_calibrator_node.cpp:277]:
```suggestion
```

File: vehicle/accel_brake_map_calibrator/src/accel_brake_map_calibrator_node.cpp [hunk 9]
```diff
@@ -482,81 +500,75 @@ void AccelBrakeMapCalibrator::callbackVelocity(const VelocityReport::ConstShared
...
     const auto past_accel_ptr =
-      getNearestTimeDataFromVec(accel_pedal_ptr_, dif_pedal_time_, accel_pedal_vec_);
+      get_nearest_time_data_from_vec(accel_pedal_ptr_, dif_pedal_time_, accel_pedal_vec_);
     const double raw_accel_pedal_speed =
-      getPedalSpeed(past_accel_ptr, accel_pedal_ptr_, accel_pedal_speed_);
+      get_pedal_speed(past_accel_ptr, accel_pedal_ptr_, accel_pedal_speed_);
     accel_pedal_speed_ = lowpass(accel_pedal_speed_, raw_accel_pedal_speed, 0.5);
-    debug_values_.data.at(CURRENT_RAW_ACCEL_SPEED) = raw_accel_pedal_speed;
-    debug_values_.data.at(CURRENT_ACCEL_SPEED) = accel_pedal_speed_;
+    debug_values_.data.at(CURRENT_RAW_ACCEL_SPEED) = static_cast<float>(raw_accel_pedal_speed);
```
Comment (shmpwk) [inline vehicle/accel_brake_map_calibrator/src/accel_brake_map_calibrator_node.cpp:528]: Great static_cast :+1:

File: vehicle/accel_brake_map_calibrator/src/accel_brake_map_calibrator_node.cpp (hunk 6 above)
Comment (N-Eiki) [reply vehicle/accel_brake_map_calibrator/src/accel_brake_map_calibrator_node.cpp:52]: I added this static_cast to handle clang-tidy warning.

File: vehicle/accel_brake_map_calibrator/src/accel_brake_map_calibrator_node.cpp (hunk 5 above)
Comment (N-Eiki) [reply vehicle/accel_brake_map_calibrator/src/accel_brake_map_calibrator_node.cpp:49]: I added this static_cast to handle clang-tidy warning.

Comment (shmpwk) [reply vehicle/accel_brake_map_calibrator/src/accel_brake_map_calibrator_node.cpp:49]: @N-Eiki 
How about 
```suggestion
  max_data_count_ = static_cast<int>(declare_parameter("max_data_count", 200));
```

File: vehicle/accel_brake_map_calibrator/src/accel_brake_map_calibrator_node.cpp (hunk 6 above)
Comment (shmpwk) [reply vehicle/accel_brake_map_calibrator/src/accel_brake_map_calibrator_node.cpp:52]: How about 
```suggestion
  precision_ = static_cast<int>(declare_parameter("precision", 3));
```

**Pull request closed without merging**